contracts/src/token/erc1155/mod.rs

```diff
@@ -0,0 +1,1973 @@
+
+            Transfer::Batch { ids, values } => receiver
+                .on_erc_1155_batch_received(
+                    call, operator, from, ids, values, data,
+                ),
+        };
+
+        let id = match result {
+            Ok(id) => id,
+            Err(e) => {
+                if let call::Error::Revert(ref reason) = e {
+                    if !reason.is_empty() {
+                        // Non-IERC1155Receiver implementer.
+                        return Err(Error::InvalidReceiverWithReason(e));
+                    }
+                }
+
+                return Err(ERC1155InvalidReceiver { receiver: to }.into());
+            }
+        };
+
+        // Token rejected.
+        if id != details.receiver_fn_selector {
+            return Err(ERC1155InvalidReceiver { receiver: to }.into());
+        }
+
+        Ok(())
+    }
+
+    /// Creates `values` of tokens specified by `ids`, and assigns
+    /// them to `to`. Performs the token acceptance check by
+    /// calling [`IERC1155Receiver::on_erc_1155_received`] or
+    /// [`IERC1155Receiver::on_erc_1155_batch_received`] on the `to` address
+    /// if it contains code.
+    ///
+    /// # Arguments
+    ///
+    /// * `&mut self` - Write access to the contract's state.
+    /// * `to` - Account of the recipient.
+    /// * `ids` - Array of all tokens ids to be minted.
+    /// * `values` - Array of all amounts of tokens to be minted.
+    /// * `data` - Additional data with no specified format, sent in call to
+    ///   `to`.
+    ///
+    /// # Errors
+    ///
+    /// If `to` is `Address::ZERO`, then the error
+    /// [`Error:InvalidReceiver`] is returned.
+    /// If length of `ids` is not equal to length of `values`, then the
+    /// error [`Error::InvalidArrayLength`] is returned.
+    /// If [`IERC1155Receiver::on_erc_1155_received`] hasn't returned its
+    /// interface id or returned with error, then the error
+    /// [`Error::InvalidReceiver`] is returned.
+    /// If [`IERC1155Receiver::on_erc_1155_batch_received`] hasn't returned its
+    /// interface id or returned with error, then the error
+    /// [`Error::InvalidReceiver`] is returned.
+    ///
+    /// # Events
+    ///
+    /// Emits a [`TransferSingle`] event if the arrays contain one element, and
+    /// [`TransferBatch`] otherwise.
+    ///
+    /// # Panics
+    ///
+    /// If updated balance exceeds `U256::MAX`, may happen during `mint`
+    /// operation.
+    fn _do_mint(
+        &mut self,
+        to: Address,
+        ids: Vec<U256>,
+        values: Vec<U256>,
+        data: &Bytes,
+    ) -> Result<(), Error> {
+        if to.is_zero() {
+            return Err(Error::InvalidReceiver(ERC1155InvalidReceiver {
+                receiver: to,
+            }));
+        }
+        self._update_with_acceptance_check(
+            Address::ZERO,
+            to,
+            ids,
+            values,
+            data,
+        )?;
+        Ok(())
```

Comment:
nit: `?` and `Ok(())` can be omitted, since `_update_with_acceptance_check`'s result (also `Ok(())`) can be propagated; same goes for a couple of other places

Don't have to change this, just noting